common/api/src/main/java/com/dfsek/terra/api/util/MathUtil.java

```diff
@@ -20,6 +18,55 @@ public final class MathUtil {
      * Epsilon for fuzzy floating point comparisons.
      */
     public static final double EPSILON = 1.0E-5;
+    private static final int SIN_BITS, SIN_MASK, SIN_COUNT;
+    private static final double radFull, radToIndex;
+    private static final double degFull, degToIndex;
+    private static final double[] sin, cos;
+    static {
+        SIN_BITS = 12;
+        SIN_MASK = ~(-1 << SIN_BITS);
+        SIN_COUNT = SIN_MASK + 1;
+        
+        radFull = Math.PI * 2.0;
+        degFull = 360.0;
+        radToIndex = SIN_COUNT / radFull;
+        degToIndex = SIN_COUNT / degFull;
+        
+        sin = new double[SIN_COUNT];
+        cos = new double[SIN_COUNT];
+        
+        for(int i = 0; i < SIN_COUNT; i++) {
+            sin[i] = Math.sin((i + 0.5f) / SIN_COUNT * radFull);
+            cos[i] = Math.cos((i + 0.5f) / SIN_COUNT * radFull);
+        }
+        
+        // Four cardinal directions (credits: Nate)
+        for(int i = 0; i < 360; i += 90) {
+            sin[(int) (i * degToIndex) & SIN_MASK] = Math.sin(i * Math.PI / 180.0);
+            cos[(int) (i * degToIndex) & SIN_MASK] = Math.cos(i * Math.PI / 180.0);
+        }
+    }
+    
+    public static double sin(double rad) {
+        return sin[(int) (rad * radToIndex) & SIN_MASK];
+    }
+    
+    public static double cos(double rad) {
+        return cos[(int) (rad * radToIndex) & SIN_MASK];
+    }
+    
+    public static double tan(double rad) {
+        return sin(rad)/cos(rad);
+    }
+    
+    public static double invSqrt(double x) {
```

Comment:
1. Funnier with comments
2Capitalize hex value

Suggested change:

```java
    public static double invSqrt(double x) {
        double halfX = 0.5d * x;
        long i = Double.doubleToLongBits(x); // evil floating point bit level hacking
        i = 0x5FE6EC85E7DE30DAL - (i >> 1); // what the fuck?
        double y = Double.longBitsToDouble(i);
        y *= (1.5d - halfX * y * y); // 1st newtonian iteration
        // y *= (1.5d - halfX * y * y); // 2nd newtonian iteration, this can be removed

        return y;
    }
```